chisel/src/executor.rs

```diff
@@ -357,6 +392,75 @@ fn format_token(token: Token) -> String {
     }
 }
 
+/// Formats a [pt::EventDefinition] into an inspection message
+///
+/// ### Takes
+///
+/// An borrowed [pt::EventDefinition]
+///
+/// ### Returns
+///
+/// A formatted [pt::EventDefinition] for use in inspection output.
+///
+/// TODO: Verbosity option
+fn format_event_definition(event_definition: &pt::EventDefinition) -> Result<String, String> {
+    let params: Result<Vec<_>, &str> = event_definition
+        .fields
+        .iter()
+        .map(|param| {
+            //
+            let kind = match &param.ty {
+                pt::Expression::Type(_, t) => match t {
+                    pt::Type::Address | pt::Type::AddressPayable | pt::Type::Payable => {
+                        ethabi::ParamType::Address
+                    }
+                    pt::Type::Bool => ethabi::ParamType::Bool,
+                    pt::Type::String => ethabi::ParamType::String,
+                    pt::Type::Int(size) => ethabi::ParamType::Int(*size as usize),
+                    pt::Type::Uint(size) => ethabi::ParamType::Uint(*size as usize),
+                    pt::Type::Bytes(len) => ethabi::ParamType::FixedBytes(*len as usize),
+                    pt::Type::DynamicBytes => ethabi::ParamType::Bytes,
+                    _ => return Err("invalid type in event definition"),
+                },
+                _ => {
+                    return Err("invalid type in event definition");
+                }
+            };
+
+            Ok(ethabi::EventParam {
+                name: match param.name.as_ref() {
+                    Some(name) => name.to_string(),
+                    None => "<anonymous>".to_string(),
+                },
+                kind,
+                indexed: param.indexed,
+            })
+        })
+        .collect();
+
+    let params = if params.is_err() {
+        return Err(format!("Could not format event definition: {}", params.unwrap_err()));
+    } else {
+        params.unwrap()
+    };
+
+    let event = ethabi::Event {
+        name: match event_definition.name.as_ref() {
+            Some(name) => name.to_string(),
+            None => "<anonymous>".to_string(),
+        },
+        inputs: params,
+        anonymous: event_definition.anonymous,
+    };
```

Comment:
- we already have a pt::Expr -> ParamType using Type
- use eyre::Result to early return

```suggestion
fn format_event_definition(event_definition: &pt::EventDefinition) -> Result<String> {
    let event_name = event_definition.name.as_ref().expect("Event has a name").to_string();
    let inputs = event_definition
        .fields
        .iter()
        .map(|param| {
            let name = param
                .name
                .as_ref()
                .map(ToString::to_string)
                .unwrap_or_else(|| "<anonymous>".to_string());
            let kind = Type::from_expression(&param.ty)
                .and_then(Type::into_builtin)
                .ok_or_else(|| eyre::eyre!("Invalid type in event {event_name}"))?;
            Ok(ethabi::EventParam { name, kind, indexed: param.indexed })
        })
        .collect::<Result<Vec<_>>>()?;

    let event = ethabi::Event { name: event_name, inputs, anonymous: event_definition.anonymous };
```